Improve loading time for UI tests loading the (default) dashboard

sgiehl opened the pull request:
### Description:

Currently the first dashboard (which is loaded by default) for UI tests contains over 30 widgets. That slows down a lot of tests, as they wait for all the widgets to be loaded.
The PR moves around the dashboards, so the first dashboard only contains a single widget.

In addition it improves the UI tests for DashboardManager to avoid full page screenshots.

refs #21458 

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

sgiehl committed:
Change default UI test fixture, so default dashboard does include only one widget

sgiehl committed:
use new dashboard ids in tests

sgiehl committed:
avoid screenshotting unneeded parts in DashboardManager UI tests

sgiehl committed:
updates expected UI test files

mneudert commented on tests/UI/specs/UIIntegration_spec.js:
```diff
@@ -61,7 +61,7 @@ describe("UIIntegrationTest", function () { // TODO: Rename to Piwik?
         this.title = parentSuite.title; // to make sure the screenshot prefix is the same
 
         it("should load dashboard1 correctly", async function () {
```
```suggestion
        it("should load dashboard2 correctly", async function () {
```

For all "should load ..." tests and probably also rename the screenshot files so they match?

michalkleiner replied:
Good spot. That would definitely trip me as recently I've been relying on the naming convention for the UI test screenshots when looking at the tests.

sgiehl committed:
Move dashboard relaated UI tests to dashboard plugin

sgiehl commented:
I've updated the PR again and cleaned up the dashboard related UI tests a bit more. The old `Dashboard` UI tests were renamed to `WidgetizedDashboard`, as they are all actually running in widgetized mode (which behaves a bit different). Also the dashboard tests that were contained in the global `UIIntegrationTests` were now moved as `Dashboard` to the dashboard plugin, where they actually should be located.

mneudert commented on plugins/Dashboard/tests/UI/Dashboard_spec.js:
```diff
@@ -6,329 +6,85 @@
+    await page.webpage.setViewport({
+      width: 1350,
+      height: 768
+    });
```
```suggestion
```

As there are no following tests this should be obsolete now. Looking at the [`app.js`](https://github.com/matomo-org/matomo/blob/2a9f9c4335801cce113dad8787fd7f85301df7ea/tests/lib/screenshot-testing/support/app.js#L322) this is probably obsolete for quite a while.

mneudert commented on plugins/Dashboard/tests/UI/WidgetizedDashboard_spec.js:
```diff
@@ -0,0 +1,334 @@
+        await page.waitForSelector('.widget');
+        await page.waitForNetworkIdle();
+
+        expect(await page.screenshot({ fullPage: true })).to.matchImage('removed');
```
This one consistently fails for me locally :thinking: 

With the change I locally get `newname` selected in first position, as I would actually expect (was at that position in both `reset` and `copied`, as per the expected screenshots).

For the expected screenshot the `newname` magically moves to the last position in the `removed` screenshot, being first place before that.  Interesting there is no CI failure and screenshot diff for that :monocle_face:

sgiehl replied:
That's really strange. The screenshot needs an update, but the test somehow doesn't fail in the action. The image processed in the action and the expected one clearly differ. Running the `compare` command on them, like the test does also returns that they differ in width or height, so there needs to be another weird reason to let the test pass nevertheless.

sgiehl committed:
apply review feedback

Co-authored-by: Marc Neudert <marc@innocraft.com>

sgiehl committed:
updates expected UI test files
